backend/dataall/modules/dataset_sharing/services/share_managers/lf_share_manager.py

```diff
@@ -432,3 +437,129 @@ def handle_revoke_failure(
+                        f'the source database {catalog_database} belongs to a catalog account {catalog_account_id}')
+            if SessionHelper.is_assumable_pivot_role(catalog_account_id):
+                self._validate_catalog_ownership_tag(catalog_account_id, catalog_region, catalog_database)
+            else:
+                raise Exception(f'Pivot role is not assumable, catalog account {catalog_account_id} is not onboarded')
+
+    def _validate_catalog_ownership_tag(self, catalog_account_id, catalog_region, catalog_database):
+        glue_client = GlueClient(account_id=catalog_account_id,
+                                 database=catalog_database,
+                                 region=catalog_region)
+
+        tags = glue_client.get_database_tags()
+        if tags.get('owner_account_id', '') == self.source_environment.AwsAccountId:
+            logger.info(
+                f'owner_account_id tag exists and matches the source account id {self.source_environment.AwsAccountId}')
+        else:
+            raise Exception(
+                f'owner_account_id tag does not exist or does not matches the source account id {self.source_environment.AwsAccountId}')
+
+    def check_catalog_account_exists_and_verify(self):
+        """
+        Checks if the source account has a catalog associated with it. This is checked by getting source catalog information and checking if there exists a target database for the source db
+        Return -
+        True - if a catalog account is present and it is verified
+        False - if no source catalog account is present
+        None - if catalog account exists but there is an issue with verifing the conditions needed for source account. Check _verify_catalog_ownership for more details
+        """
+        try:
+            catalog_dict = GlueClient(
+                account_id=self.source_environment.AwsAccountId,
+                region=self.source_environment.region,
+                database=self.dataset.GlueDatabaseName,
+            ).get_source_catalog()
+            if catalog_dict is not None:
+                # Verify the ownership of dataset by checking if pivot role is assumable and ownership tag is present
+                self._verify_catalog_ownership(catalog_dict.get('account_id'), catalog_dict.get('region'),
+                                               catalog_dict.get('database_name'))
+            else:
+                logger.info(
+                    f'No Catalog information found for dataset - {self.dataset.name} containing database - {self.dataset.GlueDatabaseName}')
+                return False
+        except Exception as e:
+            logger.error(
+                f'Failed to initialise catalog account details for share - {self.share.shareUri} '
+                f'due to: {e}'
+            )
+            return None
+        return True
+
+    def get_catalog_account_details(self):
```

Comment:
This function is a bit a duplication of `check_catalog_account_exists_and_verify`.We could have a single function:
```
def verify_and_get_catalog_account_details(self):
        try:
            catalog_dict = GlueClient(
                account_id=self.source_environment.AwsAccountId,
                region=self.source_environment.region,
                database=self.dataset.GlueDatabaseName,
            ).get_source_catalog()
            if catalog_dict is None:
                    logger.info(
                    f'No Catalog information found for dataset - {self.dataset.name} containing database - {self.dataset.GlueDatabaseName}')
            elif catalog_dict.get('account_id') != self.source_environment.AwsAccountId:
                # Verify the ownership of dataset by checking if pivot role is assumable and ownership tag is present
                self._verify_catalog_ownership(catalog_dict.get('account_id'), catalog_dict.get('region'),
                                               catalog_dict.get('database_name'))
            return catalog_dict.get('account_id', None), catalog_dict.get('region', None),
                                               catalog_dict.get('database_name', None)
        except Exception as e:
            logger.error(
                f'Failed to initialise catalog account details for share - {self.share.shareUri} '
                f'due to: {e}'
            )
            return None, None, None
        return None, None, None

```

Reply:
Initially this function was more like what you have mentioned but in order to split it and make more generic this functions was split into two . I do not mind combining it into one in favor of reducing duplication. Do you think it would be beneficial to do it this way or keep it separate so that it can be used separately and have separate responsibilities ?

Reply:
I see now where it comes from. We can leave it separate in case we want to re-use get_catalog_account_details in the future